hudi-client/hudi-spark-client/src/main/scala/org/apache/hudi/AvroConversionUtils.scala

```diff
@@ -268,4 +267,106 @@ object AvroConversionUtils {
     }
     schema
   }
+
+  /**
+   * Recursively aligns the nullable property of hoodie table schema, supporting nested structures
+   */
+  def alignFieldsNullability(sourceSchema: StructType, avroSchema: Schema): StructType = {
```

Comment:
lets please ensure each method here has unit tests.

Reply:
Try my best to add UTs for core methods as much as possible to ensure code security :)
Add ut in TestAvroConversionUtils
```
  test("test alignFieldsNullability function") {
```